legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/datasets/DataType.java

```diff
@@ -58,7 +58,10 @@
     BOOL,
     LONGTEXT,
     TEXT,
-    JSON;
+    JSON,
+    VARIANT,
```

Comment:
what is the difference between variant and json (in practical terms)?

Also, will ARRAY and MAP will be 1st class, or are these JSON?

Reply:
We had to define these types as they are supported in Snowflake. Here is what they mean:
Variant is Like Any type. It can contain normal data types or any semi structured data types.
Array is an array of Any Type
Map is like a dictionary

Map and Array are different data types in Snowflake, I agree both of them can be represented as JSON, but we will not be able to map them to individual data types in snowflake then.

Reply:
I discussed it with @davidsunglee, and I understood more this.  At some point might be good to at least add some namespace, so people understand where these are supported.